python/tskit/stats.py

```diff
@@ -163,3 +167,457 @@ def r2_matrix(self):
             A[j, j + 1 :] = a
             A[j + 1 :, j] = a
         return A
+
+
+class CoalescenceTimeDistribution:
+    """
+    Class to precompute a table of sorted/weighted node times, from which to calculate
+    the empirical distribution function and estimate coalescence rates in time windows.
+    """
+
+    class CoalescenceTimeTable:
```

Comment:
I don't think this needs to be an embedded class. Let's just pull it out into the module scope. It won't be part of the external API unless we say so, so it's fine to put it there.